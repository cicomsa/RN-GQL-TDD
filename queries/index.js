import gql from 'graphql-tag'

const GET_POSTS = gql`
  query posts {
    posts {
      id
      author
      body
    }
  }
`

const ADD_POST = gql`
  mutation addPost($input: PostInput!) {
    addPost(input: $input) {
      id
    }
  }
`


const DELETE_POST = gql`
  mutation deletePost($id: ID!) {
    deletePost(id: $id) {
      body
    }
  }
`

export {
  GET_POSTS,
  ADD_POST,
  DELETE_POST
}
