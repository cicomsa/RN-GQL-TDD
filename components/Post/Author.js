import React, { useContext } from 'react'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import { GET_POSTS, DELETE_POST } from '../../queries'
import client from '../../apollo-client'
import { PostContent } from '../../contexts'

const Author = () => {
  const { author, id } = useContext(PostContent)
  const deletePost = async id => {
    await client.mutate({
      variables: { id },
      mutation: DELETE_POST,
      refetchQueries: () => [{ query: GET_POSTS }]
    })
  }

  return (
    <TouchableOpacity className="author" activeOpacity={0.8} onPress={() => deletePost(id)}>
      <Text className="author__copy" style={styles.author}>{author}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  author: {
    color: 'purple',
    fontWeight: 'bold',
    marginVertical: 5,
    textAlign: 'center',
  }
})

export default Author