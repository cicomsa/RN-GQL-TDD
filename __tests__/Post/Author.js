import React from 'react'
import renderer from 'react-test-renderer'
import { mount, shallow } from 'enzyme'
import { View, Text } from 'react-native'
import { MockedProvider } from '@apollo/react-testing'
import wait from 'waait'
import Author from '../../components/Post/Author'
import { DELETE_POST } from '../../queries'

const deletePost = jest.fn()

describe('Post - Author', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Author />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should return View component', () => {
    const wrapper = mount(<Author />)
    expect(wrapper.find(View).exists()).toBe(true)
  })
  it('should return Text component', () => {
    const wrapper = shallow((<Author />)).dive()
    expect(wrapper.find(Text).exists()).toBe(true)
  })
  // it('should delete post', async () => {
  //   const deletePost = { author: 'Author', body: 'Body', id: 1 }
  //   let deleteMutationCalled = false
  //   const mocks = [
  //     {
  //       request: {
  //         query: DELETE_POST,
  //         variables: { id: 1 }
  //       },
  //       result: () => {
  //         deleteMutationCalled = true
  //         console.log(deleteMutationCalled)
  //         return { data: { deletePost } }
  //       }
  //     }
  //   ]

  //   const component = renderer.create(
  //     <MockedProvider mocks={mocks} addTypename={false}>
  //       <Author />
  //     </MockedProvider>
  //   )

  //   const author = component.root.findByType(View)
  //   author.props.onClick()

  //   await wait(0)
  //   console.log(deleteMutationCalled)
  //   expect(deleteMutationCalled).toBe(true)
  // })
})
