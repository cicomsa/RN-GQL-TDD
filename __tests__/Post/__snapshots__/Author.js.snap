// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post - Author should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    className="author__copy"
    style={
      Object {
        "color": "purple",
        "fontWeight": "bold",
        "marginVertical": 5,
        "textAlign": "center",
      }
    }
  >
    
  </Text>
</View>
`;
